pkg/autoscaler/autoscaler.go

```diff
@@ -57,6 +58,10 @@ type Stat struct {
 	// Lameduck indicates this Pod has received a shutdown signal.
 	// Deprecated and no longer used by newly created Pods.
 	LameDuck bool
+
+	// Average number of requests currently being handled by all ready pods of a
+	// revision.
+	AverageRevConcurrency float64
```

Comment:
I'd say with the new aggregation we don't need this anymore. You can just as well send a single metric that is aggregated in the scraper. For example:

Say the scraper has 3 samples. You can average them and send that in as the one and only metric for that revision. The autoscaler now operates on the system total anyway so it doesn't really matter how many pods send their metrics in. I think that'd get rid of a lot of the distinguishing code below and also throws out the activator branch.

Does that make sense?

Reply:
Do you mean to hidden the customer pods after scraper? In autoscaler, the metrics are either from activator or scraper. We do the aggregation for samples in scraper and send only one stat with `stat.PodName` equal to `scraper` or something similar. Is my understanding correct?

Reply:
Yeah I was thinking something along these lines.

Reply:
Make sense to me. Will do.

Reply:
Done.